### ISSUE TITLE
file input genetated via schema complex result

### ISSUE THREAD

**Issue by obalado** [wearebraid/vue-formulate#187]
**Describe the bug**
when generating file input via schema, the result is a long list of objects

**To Reproduce**
**schema:**
[
    {
    "type": "image",
    "name": "headshot",
    "label": "Your headshot"
  }
]
**Expected behavior**

{
  "headshot": [
    {
      "url": "http://via.placeholder.com/350x150.png",
      "name": "Screenshot from 2020-07-15 16-01-55.png"
    }
  ]
}
**result:**
{
  "headshot": {
    "input": {},
    "fileList": {
      "0": {}
    },
    "files": [
      {
        "progress": 100,
        "error": false,
        "complete": true,
        "justFinished": false,
        "name": "Screenshot from 2020-07-15 12-18-59.png",
        "file": {},
        "uuid": "Uu7ez-6NcEbLo5Uvey-bh",
        "path": false,
        "previewData": "data:image/png;base64," 
---SNIP---
      }
    ],
    "options": {
      "mimes": {
        "csv": "text/csv",
        "gif": "image/gif",
        "jpg": "image/jpeg",
        "jpeg": "image/jpeg",
        "png": "image/png",
        "pdf": "application/pdf",
        "svg": "image/svg+xml"
      },
      "components": {
        "FormulateSlot": {
          "inheritAttrs": false,
          "functional": true,
          "name": "FormulateSlot",
          "_Ctor": {}
        },
        "FormulateForm": {
          "name": "FormulateForm",
          "model": {
            "prop": "formulateValue",
            "event": "input"
          },
          "props": {
            "name": {
              "type": [
                null,
                null
              ],
              "default": false
            },
            "formulateValue": {},
            "values": {
              "type": [
                null,
                null
              ],
              "default": false
            },
            "errors": {
              "type": [
                null,
                null
              ],
              "default": false
            },
            "formErrors": {},
            "schema": {
              "type": [
                null,
                null
              ],
              "default": false
            }
          },
          "computed": {},
          "watch": {
            "formulateValue": {
              "deep": true,
              "user": true
            },
            "mergedFieldErrors": {
              "immediate": true,
              "user": true
            }
          },
          "methods": {},
          "staticRenderFns": [],
          "_compiled": true,
          "_Ctor": {}
        },
        "FormulateHelp": {
          "props": {
            "context": {
              "required": true
            }
          },
          "staticRenderFns": [],
          "_compiled": true,
          "name": "FormulateHelp",
          "_Ctor": {}
        },
        "FormulateLabel": {
          "props": {
            "context": {
              "required": true
            }
          },
          "staticRenderFns": [],
          "_compiled": true,
          "name": "FormulateLabel",
          "_Ctor": {}
        },
        "FormulateInput": {
          "name": "FormulateInput",
          "inheritAttrs": false,
          "inject": {
            "formulateSetter": {
              "from": "formulateSetter"
            },
            "formulateFieldValidation": {
              "from": "formulateFieldValidation"
            },
            "formulateRegister": {
              "from": "formulateRegister"
            },
            "formulateDeregister": {
              "from": "formulateDeregister"
            },
            "getFormValues": {
              "from": "getFormValues"
            },
            "validateDependents": {
              "from": "validateDependents"
            },
            "observeErrors": {
              "from": "observeErrors"
            },
            "removeErrorObserver": {
              "from": "removeErrorObserver"
            },
            "isSubField": {
              "from": "isSubField"
            }
          },
          "model": {
            "prop": "formulateValue",
            "event": "input"
          },
          "props": {
            "type": {
              "default": "text"
            },
            "name": {
              "type": [
                null,
                null
              ],
              "default": true
            },
            "formulateValue": {
              "default": ""
            },
            "value": {
              "default": false
            },
            "options": {
              "type": [
                null,
                null,
                null
              ],
              "default": false
            },
            "optionGroups": {
              "type": [
                null,
                null
              ],
              "default": false
            },
            "id": {
              "type": [
                null,
                null,
                null
              ],
              "default": false
            },
            "label": {
              "type": [
                null,
                null
              ],
              "default": false
            },
            "labelPosition": {
              "type": [
                null,
                null
              ],
              "default": false
            },
            "limit": {
              "type": [
                null,
                null
              ],
              "default": null
            },
            "minimum": {
              "type": [
                null,
                null
              ],
              "default": 0
            },
            "help": {
              "type": [
                null,
                null
              ],
              "default": false
            },
            "helpPosition": {
              "type": [
                null,
                null
              ],
              "default": false
            },
            "isGrouped": {
              "default": false
            },
            "errors": {
              "type": [
                null,
                null,
                null
              ],
              "default": false
            },
            "repeatable": {
              "default": false
            },
            "validation": {
              "type": [
                null,
                null,
                null
              ],
              "default": false
            },
            "validationName": {
              "type": [
                null,
                null
              ],
              "default": false
            },
            "error": {
              "type": [
                null,
                null
              ],
              "default": false
            },
            "errorBehavior": {
              "default": "blur"
            },
            "showErrors": {
              "default": false
            },
            "imageBehavior": {
              "default": "preview"
            },
            "uploadUrl": {
              "type": [
                null,
                null
              ],
              "default": false
            },
            "uploader": {
              "type": [
                null,
                null,
                null
              ],
              "default": false
            },
            "uploadBehavior": {
              "default": "live"
            },
            "preventWindowDrops": {
              "default": true
            },
            "showValue": {
              "type": [
                null,
                null
              ],
              "default": false
            },
            "validationMessages": {},
            "validationRules": {},
            "checked": {
              "type": [
                null,
                null
              ],
              "default": false
            },
            "disableErrors": {
              "default": false
            },
            "addLabel": {
              "type": [
                null,
                null
              ],
              "default": false
            }
          },
          "computed": {},
          "watch": {
            "$attrs": {
              "deep": true,
              "user": true
            },
            "showValidationErrors": {
              "immediate": true,
              "user": true
            }
          },
          "methods": {},
          "staticRenderFns": [],
          "_compiled": true,
          "_Ctor": {}
        },
        "FormulateErrors": {
          "inject": {
            "observeErrors": {
              "from": "observeErrors",
              "default": false
            },
            "removeErrorObserver": {
              "from": "removeErrorObserver",
              "default": false
            }
          },
          "props": {
            "context": {},
            "type": {
              "default": "form"
            }
          },
          "computed": {},
          "methods": {},
          "staticRenderFns": [],
          "_compiled": true,
          "name": "FormulateErrors",
          "_Ctor": {}
        },
        "FormulateSchema": {
          "functional": true,
          "name": "FormulateSchema",
          "_Ctor": {}
        },
        "FormulateAddMore": {
          "props": {
            "context": {
              "required": true
            },
            "addMore": {
              "required": true
            }
          },
          "staticRenderFns": [],
          "_compiled": true,
          "name": "FormulateAddMore",
          "_Ctor": {}
        },
        "FormulateGrouping": {
          "name": "FormulateGrouping",
          "props": {
            "context": {
              "required": true
            }
          },
          "inject": {
            "formulateRegisterRule": {
              "from": "formulateRegisterRule"
            },
            "formulateRemoveRule": {
              "from": "formulateRemoveRule"
            }
          },
          "computed": {},
          "watch": {},
          "methods": {},
          "staticRenderFns": [],
          "_compiled": true,
          "_Ctor": {}
        },
        "FormulateInputBox": {
          "name": "FormulateInputBox",
          "mixins": [
            {
              "props": {
                "context": {
                  "required": true
                }
              },
              "computed": {}
            }
          ],
          "staticRenderFns": [],
          "_compiled": true,
          "_Ctor": {}
        },
        "FormulateInputText": {
          "name": "FormulateInputText",
          "mixins": [
            {
              "props": {
                "context": {
                  "required": true
                }
              },
              "computed": {}
            }
          ],
          "staticRenderFns": [],
          "_compiled": true,
          "_Ctor": {}
        },
        "FormulateInputFile": {
          "name": "FormulateInputFile",
          "components": {
            "FormulateFiles": {
              "name": "FormulateFiles",
              "props": {
                "files": {
                  "required": true
                },
                "imagePreview": {
                  "default": false
                },
                "context": {
                  "required": true
                }
              },
              "computed": {},
              "watch": {},
              "staticRenderFns": [],
              "_compiled": true,
              "_Ctor": {}
            }
          },
          "mixins": [
            {
              "props": {
                "context": {
                  "required": true
                }
              },
              "computed": {}
            }
          ],
          "computed": {},
          "methods": {},
          "staticRenderFns": [],
          "_compiled": true,
          "_Ctor": {}
        },
        "FormulateRepeatable": {
          "props": {
            "context": {
              "required": true
            },
            "removeItem": {
              "required": true
            },
            "index": {
              "required": true
            }
          },
          "staticRenderFns": [],
          "_compiled": true,
          "name": "FormulateRepeatable",
          "_Ctor": {}
        },
        "FormulateInputGroup": {
          "name": "FormulateInputGroup",
          "props": {
            "context": {
              "required": true
            }
          },
          "computed": {},
          "methods": {},
          "staticRenderFns": [],
          "_compiled": true,
          "_Ctor": {}
        },
        "FormulateInputButton": {
          "name": "FormulateInputButton",
          "mixins": [
            {
              "props": {
                "context": {
                  "required": true
                }
              },
              "computed": {}
            }
          ],
          "staticRenderFns": [],
          "_compiled": true,
          "_Ctor": {}
        },
        "FormulateInputSelect": {
          "name": "FormulateInputSelect",
          "mixins": [
            {
              "props": {
                "context": {
                  "required": true
                }
              },
              "computed": {}
            }
          ],
          "computed": {},
          "staticRenderFns": [],
          "_compiled": true,
          "_Ctor": {}
        },
        "FormulateInputSlider": {
          "name": "FormulateInputSlider",
          "mixins": [
            {
              "props": {
                "context": {
                  "required": true
                }
              },
              "computed": {}
            }
          ],
          "staticRenderFns": [],
          "_compiled": true,
          "_Ctor": {}
        },
        "FormulateInputTextArea": {
          "name": "FormulateInputTextArea",
          "mixins": [
            {
              "props": {
                "context": {
                  "required": true
                }
              },
              "computed": {}
            }
          ],
          "staticRenderFns": [],
          "_compiled": true,
          "_Ctor": {}
        },
        "FormulateRepeatableRemove": {
          "props": {
            "context": {
              "required": true
            },
            "removeItem": {
              "required": true
            }
          },
          "staticRenderFns": [],
          "_compiled": true,
          "name": "FormulateRepeatableRemove",
          "_Ctor": {}
        },
        "FormulateRepeatableProvider": {
          "inject": {
            "registerProvider": {
              "from": "registerProvider"
            },
            "deregisterProvider": {
              "from": "deregisterProvider"
            }
          },
          "props": {
            "index": {
              "required": true
            },
            "context": {
              "required": true
            },
            "setFieldValue": {
              "required": true
            }
          },
          "computed": {},
          "methods": {},
          "staticRenderFns": [],
          "_compiled": true,
          "name": "FormulateRepeatableProvider",
          "_Ctor": {}
        },
        "MyFormulateAutocomplete": {
          "props": {
            "context": {
              "required": true
            }
          },
          "watch": {},
          "computed": {},
          "methods": {},
          "staticRenderFns": [],
          "_compiled": true,
          "name": "MyFormulateAutocomplete",
          "_Ctor": {}
        }
      },
      "slotComponents": {
        "label": "FormulateLabel",
        "help": "FormulateHelp",
        "errors": "FormulateErrors",
        "repeatable": "FormulateRepeatable",
        "addMore": "FormulateAddMore",
        "remove": "FormulateRepeatableRemove"
      },
      "slotProps": {},
      "library": {
        "text": {
          "classification": "text",
          "component": "FormulateInputText"
        },
        "email": {
          "classification": "text",
          "component": "FormulateInputText"
        },
        "number": {
          "classification": "text",
          "component": "FormulateInputText"
        },
        "color": {
          "classification": "text",
          "component": "FormulateInputText"
        },
        "date": {
          "classification": "text",
          "component": "FormulateInputText"
        },
        "hidden": {
          "classification": "text",
          "component": "FormulateInputText"
        },
        "month": {
          "classification": "text",
          "component": "FormulateInputText"
        },
        "password": {
          "classification": "text",
          "component": "FormulateInputText"
        },
        "search": {
          "classification": "text",
          "component": "FormulateInputText"
        },
        "tel": {
          "classification": "text",
          "component": "FormulateInputText"
        },
        "time": {
          "classification": "text",
          "component": "FormulateInputText"
        },
        "url": {
          "classification": "text",
          "component": "FormulateInputText"
        },
        "week": {
          "classification": "text",
          "component": "FormulateInputText"
        },
        "datetime-local": {
          "classification": "text",
          "component": "FormulateInputText"
        },
        "range": {
          "classification": "slider",
          "component": "FormulateInputSlider"
        },
        "textarea": {
          "classification": "textarea",
          "component": "FormulateInputTextArea"
        },
        "checkbox": {
          "classification": "box",
          "component": "FormulateInputBox"
        },
        "radio": {
          "classification": "box",
          "component": "FormulateInputBox"
        },
        "submit": {
          "classification": "button",
          "component": "FormulateInputButton"
        },
        "button": {
          "classification": "button",
          "component": "FormulateInputButton"
        },
        "select": {
          "classification": "select",
          "component": "FormulateInputSelect"
        },
        "file": {
          "classification": "file",
          "component": "FormulateInputFile"
        },
        "image": {
          "classification": "file",
          "component": "FormulateInputFile"
        },
        "group": {
          "classification": "group",
          "component": "FormulateInputGroup"
        },
        "autocomplete": {
          "classification": "text",
          "component": "MyFormulateAutocomplete"
        }
      },
      "rules": {},
      "locale": false,
      "uploadUrl": false,
      "fileUrlKey": "url",
      "uploadJustCompleteDuration": 1000,
      "plugins": [
        null,
        null
      ],
      "locales": {
        "en": {}
      },
      "idPrefix": "formulate-",
      "classes": {}
    },
    "results": [
      {
        "url": "http://via.placeholder.com/350x150.png",
        "name": "Screenshot from 2020-07-15 12-18-59.png"
      }
    ],
    "context": {
      "addLabel": "+ Your headshot",
      "attributes": {
        "id": "formulate--guide-forms-generating-forms--31",
        "name": "headshot"
      },
      "classification": "file",
      "component": "FormulateInputFile",
      "disableErrors": false,
      "errors": [],
      "formShouldShowErrors": false,
      "hasGivenName": true,
      "hasValue": false,
      "hasLabel": true,
      "help": false,
      "helpPosition": "after",
      "id": "formulate--guide-forms-generating-forms--31",
      "isValid": true,
      "imageBehavior": "preview",
      "label": "Your headshot",
      "labelPosition": "before",
      "limit": null,
      "name": "headshot",
      "minimum": 0,
      "pseudoProps": {},
      "preventWindowDrops": true,
      "repeatable": false,
      "showValidationErrors": true,
      "slotComponents": {
        "label": "FormulateLabel",
        "help": "FormulateHelp",
        "errors": "FormulateErrors",
        "repeatable": "FormulateRepeatable",
        "addMore": "FormulateAddMore",
        "remove": "FormulateRepeatableRemove"
      },
      "slotProps": {
        "label": {},
        "help": {},
        "errors": {},
        "repeatable": {},
        "addMore": {},
        "remove": {}
      },
      "type": "image",
      "uploadBehavior": "live",
      "uploadUrl": false,
      "validationErrors": [],
      "value": false,
      "visibleValidationErrors": [],
      "classes": {
        "outer": [
          "formulate-input"
        ],
        "wrapper": [
          "formulate-input-wrapper"
        ],
        "label": [
          "formulate-input-label",
          "formulate-input-label--before"
        ],
        "element": [
          "formulate-input-element",
          "formulate-input-element--image"
        ],
        "input": [],
        "help": [
          "formulate-input-help",
          "formulate-input-help--after"
        ],
        "errors": [
          "formulate-input-errors"
        ],
        "error": [
          "formulate-input-error"
        ],
        "decorator": [
          "formulate-input-element-decorator"
        ],
        "rangeValue": [
          "formulate-input-element-range-value"
        ],
        "uploadArea": [
          "formulate-input-upload-area"
        ],
        "uploadAreaMask": [
          "formulate-input-upload-area-mask"
        ],
        "files": [
          "formulate-files"
        ],
        "file": [
          "formulate-file"
        ],
        "fileName": [
          "formulate-file-name"
        ],
        "fileRemove": [
          "formulate-file-remove"
        ],
        "fileProgress": [
          "formulate-file-progress"
        ],
        "fileUploadError": [
          "formulate-file-upload-error"
        ],
        "fileImagePreview": [
          "formulate-file-image-preview"
        ],
        "fileProgressInner": [
          "formulate-file-progress-inner"
        ],
        "grouping": [
          "formulate-input-grouping"
        ],
        "groupRepeatable": [
          "formulate-input-group-repeatable"
        ],
        "groupRepeatableRemove": [
          "formulate-input-group-repeatable-remove"
        ],
        "groupAddMore": [
          "formulate-input-group-add-more"
        ]
      }
    }
  }
}

**Screenshots**

**Comment by obalado** [comment]
When files are added to a file type in Vue Formulate the value is automatically transformed into an instance of FileUpload, a helper class to wrap the FileList

object. It is recommended that the backend have a common URL where files can be uploaded.
#

**Comment by gazben** [comment]
@obalado How did you solve this issue? I get all sorts of weird error messages like:

```
Uncaught (in promise) Error: DataCloneError: Failed to execute 'replaceState' on 'History': HTMLInputElement object could not be cloned.
```
It seems that somehow there is a circular dependency that is created by the file uploader and the `file` object could not be serialized

**Comment by suhasranganath** [comment]
@obalado @gazben I am facing issues when we generate image input via Schema, especially when the image input is inside a group. Can you let us know how you were able to resolve this?

**Comment by gazben** [comment]
I only did a workaround for this. I could not solve the problem itself. What I did, is that I've created a custom component named "files" and I create a new `FormulateForm` in there with a `file` type input. I override the uploader and get the upload result from there. After I get the result from the backend I write it into the `answer` object manually and display the result in the same component with a simple list.
